Seq<VCFCompoundHeaderLine> vs. Seq<VCFHeaderLine> with javac

Attempting to call
```java
return VariantRDD.apply(
  variants.rdd(),
  SequenceDictionary.empty(),
  DefaultHeaderLines.allHeaderLines()
);
```
results in a compile error in javac, as `Seq<VCFCompoundHeaderLine>` and `Seq<VCFHeaderLine>` do not match
```
$ mvn compile
...
[ERROR] dishevelled-bio/adam/src/main/java/org/dishevelled/bio/adam/DishevelledAdamContext.java:[181,30] no suitable method found for apply(org.apache.spark.rdd.RDD<org.bdgenomics.formats.avro.Variant>,org.bdgenomics.adam.models.SequenceDictionary,scala.collection.Seq<htsjdk.variant.vcf.VCFCompoundHeaderLine>)
[ERROR]     method org.bdgenomics.adam.rdd.variant.VariantRDD.apply(org.apache.spark.sql.Dataset<org.bdgenomics.adam.sql.Variant>,org.bdgenomics.adam.models.SequenceDictionary,scala.collection.Seq<htsjdk.variant.vcf.VCFHeaderLine>) is not applicable
[ERROR]       (argument mismatch; org.apache.spark.rdd.RDD<org.bdgenomics.formats.avro.Variant> cannot be converted to org.apache.spark.sql.Dataset<org.bdgenomics.adam.sql.Variant>)
[ERROR]     method org.bdgenomics.adam.rdd.variant.VariantRDD.apply(org.apache.spark.rdd.RDD<org.bdgenomics.formats.avro.Variant>,org.bdgenomics.adam.models.SequenceDictionary,scala.collection.Seq<htsjdk.variant.vcf.VCFHeaderLine>) is not applicable
[ERROR]       (argument mismatch; scala.collection.Seq<htsjdk.variant.vcf.VCFCompoundHeaderLine> cannot be converted to scala.collection.Seq<htsjdk.variant.vcf.VCFHeaderLine>)
[ERROR] dishevelled-bio/adam/src/main/java/org/dishevelled/bio/adam/DishevelledAdamContext.java:[197,31] no suitable method found for apply(org.apache.spark.rdd.RDD<org.bdgenomics.formats.avro.Genotype>,org.bdgenomics.adam.models.SequenceDictionary,scala.collection.Seq<htsjdk.variant.vcf.VCFCompoundHeaderLine>)
[ERROR]     method org.bdgenomics.adam.rdd.variant.GenotypeRDD.apply(org.apache.spark.sql.Dataset<org.bdgenomics.adam.sql.Genotype>,org.bdgenomics.adam.models.SequenceDictionary,scala.collection.Iterable<org.bdgenomics.formats.avro.Sample>,scala.collection.Seq<htsjdk.variant.vcf.VCFHeaderLine>) is not applicable
[ERROR]       (actual and formal argument lists differ in length)
[ERROR]     method org.bdgenomics.adam.rdd.variant.GenotypeRDD.apply(org.apache.spark.rdd.RDD<org.bdgenomics.formats.avro.Genotype>,org.bdgenomics.adam.models.SequenceDictionary,scala.collection.Iterable<org.bdgenomics.formats.avro.Sample>,scala.collection.Seq<htsjdk.variant.vcf.VCFHeaderLine>) is not applicable
[ERROR]       (actual and formal argument lists differ in length)
[ERROR] -> [Help 1]
```

--- a/adam-core/src/main/scala/org/bdgenomics/adam/converters/DefaultHeaderLines.scala
+++ b/adam-core/src/main/scala/org/bdgenomics/adam/converters/DefaultHeaderLines.scala
@@ -20,6 +20,7 @@ package org.bdgenomics.adam.converters
 import htsjdk.variant.vcf.{
   VCFConstants,
   VCFFormatHeaderLine,
+  VCFHeaderLine,
   VCFHeaderLineCount,
   VCFHeaderLineType,
   VCFInfoHeaderLine,
@@ -91,7 +92,7 @@ object DefaultHeaderLines {
   /**
    * Default set of VCF header lines for INFO fields supported in ADAM.
    */
-  lazy val infoHeaderLines = Seq(
+  lazy val infoHeaderLines: Seq[VCFInfoHeaderLine] = Seq(
     ancestralAllele,
     alleleCount,
     readDepth,
@@ -169,7 +170,7 @@ object DefaultHeaderLines {
   /**
    * Default set of VCF header lines for GT format fields supported in ADAM.
    */
-  lazy val formatHeaderLines = Seq(
+  lazy val formatHeaderLines: Seq[VCFFormatHeaderLine] = Seq(
     genotype,
     genotypeQuality,
     allelicDepth,
@@ -188,5 +189,5 @@ object DefaultHeaderLines {
   /**
    * Default set of VCF header lines for INFO and GT format fields supported in ADAM.
    */
-  lazy val allHeaderLines = infoHeaderLines ++ formatHeaderLines
+  lazy val allHeaderLines: Seq[VCFHeaderLine] = infoHeaderLines ++ formatHeaderLines
 }

